kernel/thread.c

```diff
@@ -1210,14 +1210,19 @@ size_t myst_kill_thread_group()
     myst_spin_lock(process->thread_lock);
     for (t = tail; t != NULL; t = t->group_prev)
     {
-        //        if (!myst_is_process_thread(t) && t->status == MYST_RUNNING)
         if (t != thread && t->status == MYST_RUNNING)
         {
             count++;
             myst_spin_unlock(process->thread_lock);
             myst_signal_deliver(t, SIGKILL, 0);
+
             // Wake up the thread from futex_wait if necessary.
-            myst_cond_signal_thread(t->signal.cond_wait, t);
+            if (t->signal.waiting_on_event)
```

Comment:
Can we move this to `myst_signal_deliver`? That would cover waking futexes in non-shutdown scenarios, like musl SIGSYNCCALL.
However, in the non-shutdown case there is also the issue of restarting syscalls if SA_RESTART is set.

Reply:
The last time I tried that, dotnet-proc-maps hangs for some reasons. I will give it another try.

I am not `SA_RESTART` applies when a kill signal interrupts the operation.